packages/auth/amplify_auth_cognito/lib/src/state/machines/auth_state_machine.dart

```diff
@@ -37,7 +37,15 @@ class AuthStateMachine extends AuthStateMachineBase {
   /// The AWS credentials provider, using cached credentials.
   late final AWSCredentialsProvider credentialsProvider =
       InlineCredentialsProvider(() async {
-    throw UnimplementedError();
+    // Get the latest credentials from storage.
+    dispatch(const FetchAuthSessionEvent.fetch());
+    final fetchState =
+        await expect(FetchAuthSessionStateMachine.type).getLatestResult();
+    final fetchedCredentials = fetchState?.session.credentials;
```

Comment:
minor nit/idea: would it be better to encapsulate the FetchAuth functionality into a separate class taht would handle the dispatch/expect taking of credentials instead of having to directly reference the different FetchAuth components directly from within auth_state_machine?

Reply:
To an external observer, it is a single state machine. The goal is to abstract away the fact that it manages multiple state machines because all communication should happen with this state machine.

Reply:
Great idea! I've extracted this into `AuthPluginCredentialsProvider`. This will make it easier to test now as well :)